### PR TITLE
Default OpenAL Audio on Windows

### PR DESCRIPTION
**Pull request by RobertBColton**
I am doing this because I get tired of having to change it all the time, the DirectSound system is just not a good default.

* DirectSound crashes on MP3s and anything other than WAVs.
* DirectSound has crappy support for mixing and positional sound.
* GMS encourages people to use the newer audio API which makes it more difficult to just run stuff.

### CODE REVIEW

**Comment by fundies** [comment]
Issue with this is openal requires many dlls to be shipped with your game. We can automate the process of shipping the dlls with cmake but we havent migrated the build system yet. I did start it in #1490 but it was never merged

**Comment by RobertBColton** [comment]
@fundies Nevermind, the answer isn't using OpenAL on Windows, the answer is to fix DirectSound and add XAudio2 as I've been planning. There's quite a bit to this but basically, it goes back to me telling you and Josh to convert all sounds into OGG before tacking them onto the exe (like how we turn textures all into 32 bit bgra). One of the reasons to do this, in addition to getting around the OpenAL linking issues, is the MP3 license. YoYoGames had to do this recently due to LGPL license problems.
https://www.reddit.com/r/gamedev/comments/8l23h6/lgpl_violation_in_games_exported_with_game_maker/

Nobody was sure what they did, I can't find it on the internet. However, from testing, I see that with "Use New Audio Engine" checked, I still get DirectSound, even when I am calling the positional audio sounds. I believe what YoYo did was just mimic OpenAL using their existing DirectSound system (and I know how to do what they did via [DuplicateSoundBuffer](https://docs.microsoft.com/en-us/previous-versions/windows/desktop/ee418041(v=vs.85))).
GMS New Audio Engine DirectSound

So basically, to fix this sounds mess, we keep using OpenAL on Linux as usual. And you and Josh either allow me to use an OGG/MP3 library, I find out how to load it with media engine, or you and Josh decide to convert all audio files to OGG similar to how we do images. The last thing might be required in the case of XAudio2/UWP/Xbox anyway.

**Comment by RobertBColton** [comment]
@fundies to further my point, you don't _have_ the Windows media engine available on Xbox, so how would you load MP3 and stuff? That's why you have to convert to a single format, like we do for images 32 bit bgra.
